include/cyclic_reduction.hxx

```diff
@@ -421,122 +424,92 @@ private:
     //     from each processor. The number of systems of equations received will
     //     vary from myns to myns+1 (if myproc >= nsextra).
     // The size of the array reserved is therefore (myns+1)
+    recvbuffer = Matrix<T>(nprocs, (myns+1)*8); // Buffer for receiving from other processors
     
-    recvbuffer = matrix<T>(nprocs, (myns+1)*8); // Buffer for receiving from other processors
-    
-    if (myns > 0) {
-    
-      // Some interface systems to be solved on this processor
-      // Note that the interface equations are organised by system (myns as first argument)
-      // but communication buffers are organised by processor (nprocs first).
-      
-      ifcs = matrix<T>(myns, 2*4*nprocs);     // Coefficients for interface solve. 
-      if (nprocs > 1) {
-        if2x2 = matrix<T>(myns, 2*4);         // 2x2 interface equations on this processor
-      }
-      ifx  = matrix<T>(myns, 2*nprocs);       // Solution of interface equations.
-      // Each system to be solved on this processor has two interface equations from each processor
-      
-      ifp = new T[myns*2];     // Solution to be sent to processor p
-      
-    }
-    
-    x1 = new T[Nsys];
-    xn = new T[Nsys];
+    // Some interface systems to be solved on this processor
+    // Note that the interface equations are organised by system (myns as first argument)
+    // but communication buffers are organised by processor (nprocs first).
+    ifcs = Matrix<T>(myns, 2*4*nprocs);     // Coefficients for interface solve
+    if(nprocs > 1)
+      if2x2 = Matrix<T>(myns, 2*4);         // 2x2 interface equations on this processor
+    ifx = Matrix<T>(myns, 2*nprocs);       // Solution of interface equations
+    ifp = Array<T>(myns*2);     // Solution to be sent to processor p
+    // Each system to be solved on this processor has two interface equations from each processor
     
+    x1 = Array<T>(Nsys);
+    xn = Array<T>(Nsys);
   }
 
   /// Free all memory arrays allocated by allocMemory()
   void freeMemory() {
```

Comment:
Is this needed at all now? Can it be deprecated/removed?

Reply:
It currently also resets the problem size. I don't think we really need this as it should be cheap enough to fully tear down and rebuild a new instance should we need to. May be worth deprecating?

Reply:
I've marked as deprecated, but this is private so I think we should probably be free to just remove it if we decide that we don't need to reset the problem size or if we want to do this a different way.